beginner_source/blitz/tensor_tutorial.py

```diff
@@ -1,195 +1,189 @@
 
-x = torch.tensor([5.5, 3])
-print(x)
+new_ones_tnsr = torch.ones_like(tnsr_from_data) # 2 x 2 matrix of ones
 
-###############################################################
-# or create a tensor based on an existing tensor. These methods
-# will reuse properties of the input tensor, e.g. dtype, unless
-# new values are provided by user
+try:
+  new_rand_tnsr = torch.rand_like(tnsr_from_data) # 2 x 2 matrix of random numbers
+except RuntimeError as e:
+  print(f"RuntimeError thrown: {e}")
+  print()
+  print(f"Random values in PyTorch are floating points. Datatype passed to torch.rand_like(): {tnsr_from_data.dtype}")
 
-x = x.new_ones(5, 3, dtype=torch.double)      # new_* methods take in sizes
-print(x)
 
-x = torch.randn_like(x, dtype=torch.float)    # override dtype!
-print(x)                                      # result has the same size
+######################################################################
+# This works after we override the dtype property
+#
 
-###############################################################
-# Get its size:
+new_rand_tnsr = torch.rand_like(tnsr_from_data, dtype=torch.float)
+print(new_rand_tnsr)
 
-print(x.size())
 
-###############################################################
-# .. note::
-#     ``torch.Size`` is in fact a tuple, so it supports all tuple operations.
+######################################################################
+# **With random or constant values:**
 #
-# Operations
-# ^^^^^^^^^^
-# There are multiple syntaxes for operations. In the following
-# example, we will take a look at the addition operation.
-#
-# Addition: syntax 1
-y = torch.rand(5, 3)
-print(x + y)
 
-###############################################################
-# Addition: syntax 2
+shape = (2,3,)
+rand_tnsr = torch.rand(shape)
+ones_tnsr = torch.ones(shape)
+zeros_tnsr = torch.zeros(shape)
+print(f"Random Tensor:\n{rand_tnsr} \n\n \
+        Ones Tensor:\n{ones_tnsr} \n\n \
+        Zeros Tensor:\n{zeros_tnsr}")
 
-print(torch.add(x, y))
 
-###############################################################
-# Addition: providing an output tensor as argument
-result = torch.empty(5, 3)
-torch.add(x, y, out=result)
-print(result)
 
-###############################################################
-# Addition: in-place
 
-# adds x to y
-y.add_(x)
-print(y)
+######################################################################
+# --------------
+#
 
-###############################################################
-# .. note::
-#     Any operation that mutates a tensor in-place is post-fixed with an ``_``.
-#     For example: ``x.copy_(y)``, ``x.t_()``, will change ``x``.
+
+######################################################################
+# Tensor Attributes
+# ~~~~~~~~~~~~~~~~~
 #
-# You can use standard NumPy-like indexing with all bells and whistles!
+# Tensor attributes describe their shape data type and where they live.
```

Comment:
need commas after shape and data type - "where they live" makes sense to me but I worry it might not translate well... maybe "on which device the tensor is stored" or something similar.